src/applications/oscilloscope.py

```diff
@@ -177,39 +333,55 @@ def on_closing(self):
             logger.debug(e)
             pass
 
+
 def build_data_model():
+    """ Builds the data acquisition model based on arguments. """
     if args.randomtest:
         data_acquisition_model = datasources.RandomRateCounter()
     else:
-        data_acquisition_model = datasources.NiDaqDigitalInputRateCounter(daq_name = args.daq_name,
-                                                                          signal_terminal = args.signal_terminal,
-                                                                          clock_rate = args.clock_rate,
-                                                                          num_data_samples_per_batch = args.num_data_samples_per_batch,
-                                                                          clock_terminal = args.clock_terminal,
-                                                                          read_write_timeout = args.rwtimeout,
-                                                                          signal_counter = args.signal_counter)
+        data_acquisition_model = datasources.NiDaqDigitalInputRateCounter(
+            daq_name=args.daq_name,
+            signal_terminal=args.signal_terminal,
+            clock_rate=args.clock_rate,
+            num_data_samples_per_batch=args.num_data_samples_per_batch,
+            clock_terminal=args.clock_terminal,
+            read_write_timeout=args.rwtimeout,
+            signal_counter=args.signal_counter
+        )
     return data_acquisition_model
 
-def run_console():
 
-    view = ScopeFigure(args.scope_width)
+def run_console():
+    """ Runs only the matplotlib animation, no sidebar. """
+    rolling_mean_window_size = np.ceil(args.rolling_mean_window / args.animation_update_interval)
+    view = ScopeFigure(args.scope_width, int(rolling_mean_window_size))
```

Comment:
There's an error when rolling_mean_window_size is greater than scope_width. Probably need to check this somewhere and display a message to the user, then set rolling_mean_window_size == scope_width.